packages/yarnpkg-core/sources/StreamReport.ts

```diff
@@ -189,6 +192,7 @@ export class StreamReport extends Report {
     includeWarnings = includeLogs,
     forgettableBufferSize = BASE_FORGETTABLE_BUFFER_SIZE,
     forgettableNames = new Set(),
+    enableProgressBars = configuration.get(`enableProgressBars`) && !json && (stdout as WriteStream).isTTY && (stdout as WriteStream).columns > 22,
```

Comment:
Do we need to make this configurable? It feels like it should be the stream's responsibility, not the stream's caller - for example, this logic causes weird behavior where the progress bar would be displayed even with `--json` if the caller set `enableProgressBar` explicitly to true (same thing for the screen width check, which the caller shouldn't have to worry about).